interfaces/cython/cantera/with_units/solution.py.in

```
@@ -0,0 +1,224 @@
+    a custom name can assigned via::
+
+        gas.name = 'my_custom_name'
+
+    In addition, `Solution` objects can be constructed by passing the text of
+    the YAML phase definition in directly, using the ``yaml`` keyword
+    argument::
+
+        yaml_def = '''
+        phases:
+        - name: gas
+          thermo: ideal-gas
+          kinetics: gas
+          elements: [O, H, Ar]
+          species:
+          - gri30.yaml/species: all
+          reactions:
+          - gri30.yaml/reactions: declared-species
+          skip-undeclared-elements: true
+          skip-undeclared-third-bodies: true
+          state: {T: 300, P: 1 atm}
+        '''
+        gas = ct.Solution(yaml=yaml_def)
+    """
+    def __init__(self, infile="", name="", *, yaml=None):
+        self.__dict__["_phase"] = _Solution(infile, name, yaml=yaml)
+
+    @common_properties@
+
+    @thermophase_properties@
+
+
+class PureFluid:
+    """
+    This implementation of `PureFluid <cantera.with_units.PureFluid>` operates with
+    units by using the `pint` library to convert between unit systems. All properties
+    are asssigned units in the standard MKS system that Cantera uses, substituting kmol
+    instead of mol. Each property is an instance of the `pint.Quantity` class.
+
+    Similarly, properties must be instances of `pint.Quantity` classes when they are
+    used for assignment to set the state. The properties may have any units, so long
+    as the dimensions for the quantity are consistent. For example, temperatures can
+    be provided in K, degC, degF, or degR; conversion will be done internally to
+    Cantera's consistent unit system.
+
+    See the `pint documentation <https://pint.readthedocs.io>`__ for more information
+    about using pint's ``Quantity`` classes.
+    """
+    def __init__(self, infile, name="", *, yaml=None, **kwargs):
+        self.__dict__["_phase"] = _PureFluid(infile, name, **kwargs)
```

Silently ignoring the `yaml` argument doesn't sound right -- I assume it can just be removed here, and let it propagate to the `_PureFluid` constructor, where I believe it's a valid (if awkward) way of initializing.